pdns/lwres.cc

```diff
@@ -308,16 +310,23 @@ LWResult::Result asyncresolve(const ComboAddress& ip, const DNSName& domain, int
     }
 
     ret = asendto((const char*)&*vpacket.begin(), vpacket.size(), 0, ip, qid,
-                  domain, type, &queryfd);
+                  domain, type, &queryfd, fstrmQEnabled ? &clientip : nullptr);
 
```

Comment:
I wonder if it would not make more sense to call `getsockname()` on the returned `queryfd` from here instead of doing it in `asendto`. Or if we don't care about altering more code for that feature, perhaps skip that call completely by going all the way down to `makeClientSocket()`, but that does not seem very clean.

Reply:
We're looking at remains of an earlier attempt to set the address in `makeClientSocket()`. I then decided to push the `getsockname()` call upward for cleanliness, but I see now that it can be pushed even higher.